crates/variables/src/provider/vault.rs

```diff
@@ -24,12 +25,14 @@ impl VaultProvider {
         token: impl Into<String>,
         mount: impl Into<String>,
         prefix: Option<impl Into<String>>,
+        retry_limit: Option<u8>,
```

Comment:
Is there a meaningful semantic difference between `None` and `Some(0)`? Seems like we could just pass a `u8` no?